### ISSUE TITLE
"Transaction rolled back because it has been marked as rollback-only" in some cases

### ISSUE THREAD

**Issue by dcoraboeuf** [nemerosa/ontrack#424]
In version 2.20.8, when getting a filtered view, there is the following exception:

```
org.springframework.transaction.UnexpectedRollbackException: Transaction rolled back because it has been marked as rollback-only
        at org.springframework.transaction.support.AbstractPlatformTransactionManager.commit(AbstractPlatformTransactionManager.java:724)
        at org.springframework.transaction.interceptor.TransactionAspectSupport.commitTransactionAfterReturning(TransactionAspectSupport.java:485
)
        at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:291)
        at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:96)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
        at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:655)
        at net.nemerosa.ontrack.service.StructureServiceImpl$$EnhancerBySpringCGLIB$$26dd11db.getBuildView(<generated>)
        at net.nemerosa.ontrack.boot.ui.BranchController.lambda$buildViewWithFilter$42(BranchController.java:478)
        at net.nemerosa.ontrack.boot.ui.BranchController$$Lambda$185/956625328.apply(Unknown Sourc
```

No more detail is available and investigation must be done at the customer.

**Comment by dcoraboeuf** [comment]
Root cause: there is regex pattern parsing failure in `StructureServiceImpl#buildSearch` and the thrown exception will make the transaction rolled back.

This issue is two-folded:

* this code should be protected since the regex being matched is ultimately entered by an end-user
* in this very case, this is the `net.nemerosa.ontrack.extension.general.BuildLinkPropertyItem` which calls the search, based on an item whose content was created after an (other) error:

![image](https://cloud.githubusercontent.com/assets/1206964/15715934/b290e9c0-2820-11e6-89c5-b8024d0d7f84.png)

So the real root cause lies in another issue, in the build of the customer, where the build name being entered is totally wrong.

The list of actions is:

1. protect the code mentioned against wrong regex expressions
1. remove the faulty link manually
1. to correct the code which created the faulty link

**Comment by dcoraboeuf** [comment]
Fixed in 2.21.2
